docs/developing.md

```diff
@@ -57,3 +57,47 @@ $ cd <application project directory>
 $ yalc remove @hypothesis/frontend-shared
 $ yarn --check-files
 ```
+
+## Adding new components
+
+Before adding a component, determine what category of component it will be — simple, presentational or composite (details can be found in the Pattern Library) — and what group of components it belongs with (e.g. Layout, Data Display).
+
+Each component in this package should have:
+
+1. A component module in `src/components`
+2. A test module
+3. Pattern-library documentation in `src/pattern-library/components`
+4. Exports from its module directory and the project's entrypoint
+
+### Generating component module files
+
+You may create modules manually, or you can use [`plop`](https://plopjs.com/) to scaffold out some starter files. Run:
+
+```sh
+$ yarn run plop
+```
+
+and follow the prompts.
+
+### New component checklist
+
+- [ ] determine component category (simple, presentational, or composite)
+- [ ] implement the component
+- [ ] write tests for the component
+- [ ] verify that exports are appropriate
+- [ ] test against an external application (`client` or `lms`) as described in "Testing locally with other projects"
```

Comment:
It would be good to make "Testing locally with other projects" a link into docs/developing.md here.

Reply:
We're already _in_ `developing.md` 😛